### PR TITLE
Add postgres

### PR DESCRIPTION
**Pull request by liufuyang**
image
---

image
---

image
---

image
---

### CODE REVIEW

File: backend/resources/user_resources.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    @jwt_required()
+    def get(self, uid):
+        user = UserModel.find_by_uid(uid)
+        if user:
+            # no need to use jsonify as flask_restful does that for us
+            return user.json()
+        else:
+            return {'message': 'User does not exist'}, 404  # not found
+
+    def post(self, uid):
```
Comment (sheepray) [inline backend/resources/user_resources.py:18]: Do we also need *@jwt_required()* in here?

File: backend/resources/user_resources.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        user = UserModel(uid, request_data['password'])
+        
+        try:
+            user.save_to_db()
+        except:
+            return {"message": "An error occurred inserting the item."}, 500
+
+        return user.json(), 201
+
+    def put(self, uid):
```
Comment (sheepray) [inline backend/resources/user_resources.py:35]: Same as above.

File: backend/security.py [hunk 3]
```diff
@@ -1,23 +1,14 @@
...
-                             password.encode('utf-8')):
+    user = UserModel.find_by_uid(uid)
+    if user and safe_str_cmp(user.password, password):
         return user
 
 
 def identity(payload):
     uid = payload['identity']
-    return uid_mapping.get(uid, None)
+    print(payload)
```
Comment (sheepray) [inline backend/security.py:13]: debug purpose?

**Review by sheepray — APPROVED** [review]

File: backend/resources/user_resources.py (hunk 1 above)
Comment (liufuyang) [reply backend/resources/user_resources.py:18]: yes, when the API starts to work we can always add it. for now we don't have to worry making all API perfect. I left post and put out for auth so to make it easier to test. I think we can add the tag later on.

File: backend/security.py (hunk 3 above)
Comment (liufuyang) [reply backend/security.py:13]: yes, I will remove it.

Comment (liufuyang) [reply backend/security.py:13]: it seems this jwt identity thing can only work if our user data model have a field called `id`. I remember in the beginning I tried to call the filed as `uid` in user model and database, and this identity gives error.

**Pull request merged**